File: programmers/sort/BiggestNumber.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        StringBuilder sb = new StringBuilder();

        String[] stringNumbers = new String[numbers.length];
        for (int i = 0; i < stringNumbers.length; i++) {
            stringNumbers[i] = Integer.toString(numbers[i]);
        }

        Arrays.sort(stringNumbers, new Comparator<String>() {
            @Override
            public int compare(String o1, String o2) {
                return Integer.parseInt(o2+o1) - Integer.parseInt(o1+o2);
            }
        });

        for (String number : stringNumbers) {
            sb.append(number);
        }

        String answer = sb.toString();
        if (answer.charAt(0) == '0') answer = "0";

        return answer;
    }
}

File: programmers/sort/BiggestNumber.kt
class BiggestNumber {
    fun solution(numbers: IntArray): String {
        var answer = ""
        val numbersList = numbers.map { Number(it.toString()) }
        numbersList.forEach { number: Number ->
            var priority = number.string
            while (3 < priority.length) {
                priority += number.string
            }
            number.priority = priority
        }
        numbersList.sortedBy { it.priority }.reversed()
        numbersList.forEach { number ->
            answer += number.string
        }
        return answer
    }

    data class Number(val string: String) {
        lateinit var priority: String
    }
}